src/Mvc/Extensions.ApiDescription.Client/src/build/Microsoft.Extensions.ApiDescription.Client.targets

```diff
@@ -0,0 +1,151 @@
+        OutputDirectory="$(OpenApiDefaultOutputDirectory)">
+      <Output TaskParameter="Outputs" ItemName="_Temporary" />
+    </GetFileReferenceMetadata>
+
+    <ItemGroup>
+      <OpenApiReference Remove="@(OpenApiReference)" />
+      <OpenApiReference Include="@(_Temporary)" />
+      <_Temporary Remove="@(_Temporary)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_GetCurrentOpenApiReference">
+    <GetCurrentItems Input="$(GeneratorMetadata)">
+      <Output TaskParameter="Outputs" ItemName="CurrentOpenApiReference" />
+    </GetCurrentItems>
+  </Target>
+
+  <Target Name="_InnerGenerateOpenApiReferenceCode" DependsOnTargets="_GetCurrentOpenApiReference;$(GeneratorTarget)" />
+
+  <Target Name="_GenerateOpenApiReferenceCode" Inputs="@(OpenApiReference)" Outputs="%(OutputPath)"
+      Condition="$(OpenApiGenerateAtDesignTime) OR ('$(DesignTimeBuild)' != 'true' AND '$(BuildingProject)' == 'true') OR !Exists('%(OutputPath)')">
+    <MSBuild BuildInParallel="$(BuildInParallel)"
+        Projects="$(MSBuildProjectFullPath)"
+        Properties="GeneratorTargetPath=%(OpenApiReference.OutputPath);GeneratorTarget=Generate%(CodeGenerator);GeneratorMetadata=%(SerializedMetadata)"
+        RemoveProperties="TargetFrameworks"
+        Targets="_InnerGenerateOpenApiReferenceCode" />
+  </Target>
+
+  <Target Name="_CreateCompileItemsForOpenApiReferences" Condition="'@(OpenApiReference)' != ''">
+    <ItemGroup>
+      <_Files Remove="@(_Files)" />
+      <_Files Include="@(OpenApiReference -> '%(OutputPath)')"
+          Condition="$([System.IO.File]::Exists('%(OpenApiReference.OutputPath)'))">
+        <OutputPathExtension>$([System.IO.Path]::GetExtension('%(OpenApiReference.OutputPath)'))</OutputPathExtension>
+      </_Files>
+      <_Directories Remove="@(_Directories)" />
+      <_Directories Include="@(OpenApiReference -> '%(OutputPath)')"
+          Condition="Exists('%(OpenApiReference.OutputPath)') AND ! $([System.IO.File]::Exists('%(OpenApiReference.OutputPath)'))" />
+
+      <!-- If OutputPath is a file, add it directly to relevant items. -->
+      <TypeScriptCompile Remove="@(_Files)" Condition="'%(_Files.OutputPathExtension)' == '.ts'" />
+      <TypeScriptCompile Include="@(_Files)" Condition="'%(_Files.OutputPathExtension)' == '.ts'">
+        <SourceDocument>%(_Files.FullPath)</SourceDocument>
+      </TypeScriptCompile>
+
+      <Compile Remove="@(_Files)"
+          Condition="'$(DefaultLanguageSourceExtension)' != '.ts' AND '%(_Files.OutputPathExtension)' == '$(DefaultLanguageSourceExtension)'" />
+      <Compile Include="@(_Files)"
+          Condition="'$(DefaultLanguageSourceExtension)' != '.ts' AND '%(_Files.OutputPathExtension)' == '$(DefaultLanguageSourceExtension)'">
+        <SourceDocument>%(OpenApiReference.FullPath)</SourceDocument>
+      </Compile>
+
+      <!-- Otherwise, add all descendant files with the expected extension. -->
+      <TypeScriptCompile Remove="@(_Directories -> '%(Identity)/**/*.ts')" />
```

Comment:
I'm assuming you're doing the `Remove` to avoid duplicate entries, right? A more efficient way to do this is in MSBuild is
```xml
<TypeScriptCompile Include=" new pattern here " Exclude="@(TypeScriptCompile)" />
```